x/mint/keeper/keeper.go

```diff
@@ -151,8 +165,30 @@ func (k Keeper) MintCoins(ctx sdk.Context, newCoins sdk.Coins) error {
 	return k.bankKeeper.MintCoins(ctx, types.ModuleName, newCoins)
 }
 
+// GetPoolAllocatableAsset gets the balance of the `MintedDenom` from the `feeCollectorName` module account and returns coins according to the `AllocationRatio`
+func (k Keeper) GetPoolAllocatableAsset(ctx sdk.Context) sdk.Coin {
+	params := k.GetParams(ctx)
+
+	feeCollector := k.accountKeeper.GetModuleAccount(ctx, k.feeCollectorName)
+	asset := k.bankKeeper.GetBalance(ctx, feeCollector.GetAddress(), params.MintDenom)
+
+	return sdk.NewCoin(asset.Denom, asset.Amount.ToDec().Mul(params.PoolAllocationRatio).TruncateInt())
+}
+
 // AddCollectedFees implements an alias call to the underlying supply keeper's
 // AddCollectedFees to be used in BeginBlocker.
 func (k Keeper) AddCollectedFees(ctx sdk.Context, fees sdk.Coins) error {
-	return k.bankKeeper.SendCoinsFromModuleToModule(ctx, types.ModuleName, k.feeCollectorName, fees)
+	err := k.bankKeeper.SendCoinsFromModuleToModule(ctx, types.ModuleName, k.feeCollectorName, fees)
+
+	// allocate pool allocation ratio to pool-incentives module account account
+	coins := sdk.NewCoins(k.GetPoolAllocatableAsset(ctx))
+	err = k.bankKeeper.SendCoinsFromModuleToModule(ctx, k.feeCollectorName, poolincentivestypes.ModuleName, coins)
+	if err != nil {
+		return err
+	}
+
```

Comment:
Where should we distribute remaining coins?  If fee collector get increased as time goes, it should get infinite 20% of those staked fees?
80% of fee will be remaining at first.
20% of (previous 80% + new mint amount) will be released next time.
Is this correct logic? I think we need to only provide 20% of new mint amount.

Reply:
Here's the Draft PR for possible modifications on my thoughts.
https://github.com/osmosis-labs/osmosis/pull/170/files

Reply:
There should be no leftover amount of minted tokens right? Like we should setup our distribution such that it all gets cleared

(Also we shouldn't be calling this fees, tx fees should really only be going to stakers. Lets call it `mintedCoins` perhaps?)

Reply:
Nothing is left after new implementation